core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java

```diff
@@ -123,6 +164,38 @@ public void write(byte[] b, int off, int len) throws IOException {
     updateCurrentPacket(false);
   }
 
+  /**
+   * Writes the data in the specified byte buf to this output stream.
+   *
+   * @param buf the buffer
+   * @throws IOException
+   */
+  public void write(io.netty.buffer.ByteBuf buf) throws IOException {
+    write(buf, 0, buf.readableBytes());
+  }
+
+  /**
+   * Writes len bytes from the specified byte buf starting at offset off to this output stream.
+   *
+   * @param buf the buffer
+   * @param off the offset
+   * @param len the length
+   */
+  public void write(io.netty.buffer.ByteBuf buf, int off, int len) throws IOException {
+    if (len == 0) {
+      return;
+    }
+
+    while (len > 0) {
+      updateCurrentPacket(false);
+      int toWrite = Math.min(len, mCurrentPacket.writableBytes());
```

Comment:
If mCurrentPacket.writableBytes() < len, shouldn't we come out of the loop after the write ?